pkg/utils/errors/errors.go

```diff
@@ -116,15 +118,15 @@ func GetID(err error) string {
 type ErrorContext struct {
 	name         string
 	lastErrorIDs []string
-	errorIDs     map[string]struct{}
```

Comment:
It's better not to use sets here, as it makes this package dependent on `k8s.io/apimachinery`, any package that doesn't need k8s dependency and wants to use this package will have to import  `k8s.io/apimachinery/pkg/util/sets`, which is not ideal IMO